### ISSUE TITLE
CA2000 should report on JsonDocument instances returned from JsonDocument.Parse

### ISSUE THREAD

**Issue by stephentoub** [dotnet/roslyn-analyzers#6861]
JsonDocument is disposable, and it's unfortunately common to see code that things like:
```C#
JsonElement json = JsonDocument.Parse(...).RootElement;
```
CA2000 should warn that the JsonDocument instance returned from Parse needs to be disposed.

**Comment by Alex-Sob** [comment]
Hello .NET team,

I have some background in writing C# analyzers for Microsoft projects.
I would like to try fixing this. It seems that it's not reporting an error in a simpler case as well:

`var doc = JsonDocument.Parse("{}");`

**Comment by himanshusaini111** [comment]
Yes its not wroking for simpler cases too. And this issue just reminded me of another memory leak I created in My system with JsonDocument.Parse 😅

Anyways for it was not working for HttpResponseMessage objects in cases like:

`var response = await httpClient.GetAsync("api/get-beans");`

Now atleast I know why. Can not simply check the return type of a Function? Atleast it should be fixed for all the Functions Available in .Net sdk itself.
